packages/p2p/src/components/my-ads/form-ads.jsx

```diff
@@ -32,32 +41,56 @@ class FormAds extends Component {
         );
         const display_residence = ObjectUtils.getPropertyValue(current_residence, 'text') || '';
 
-        // TODO: [p2p-fix-api] call get offer detail api and populate state
-        // this is pending offer edit enabled in the API
-        if (this.props.ad_id) {
-            // call the api, get the file based on id
-            // populate the state from the respnose
+        const { data } = this.props;
+
+        // user is editing an existing ad, we should populate its data
+        if (this.isUpdatingAd()) {
             this.setState({
-                country: display_residence,
+                is_loading      : false,
+                advertiser_notes: data.advertiser_notes,
+                country         : display_residence,
+                currency        : data.transaction_currency,
+                max_transaction : data.max_transaction,
+                min_transaction : data.min_transaction,
+                offer_amount    : data.offer_amount,
+                offer_currency  : data.offer_currency,
+                payment_method  : data.payment_method,
+                price_rate      : data.price_rate,
+                type            : data.type,
             });
         } else {
-            this.setState({ is_loading: false, country: display_residence });
+            this.setState({
+                is_loading    : false,
+                country       : display_residence,
+                currency      : this.context.local_currency_config.currency,
+                offer_currency: this.context.currency,
+            });
         }
     }
 
+    isUpdatingAd = () => !ObjectUtils.isEmptyObject(this.props.data);
```

Comment:
we can put this in state as variable instead of function

Reply:
@mustofa-binary This is simply a getter, doesn't need to be in state. Plus, putting props in state is a "less-than-ideal" because it breaks the synchronisation between view and props.